File: app/src/main/java/com/raival/fileexplorer/activity/model/MainViewModel.java
package com.raival.fileexplorer.activity.model;

import androidx.lifecycle.ViewModel;

import com.raival.fileexplorer.tab.BaseDataHolder;
import com.raival.fileexplorer.tab.file.model.Task;

import java.util.ArrayList;
import java.util.List;

public class MainViewModel extends ViewModel {
    public final ArrayList<Task> tasks = new ArrayList<>();
    private final List<BaseDataHolder> dataHolders = new ArrayList<>();

    public void addDataHolder(BaseDataHolder dataHolder) {
        dataHolders.add(dataHolder);
    }

    public List<BaseDataHolder> getDataHolders() {
        return dataHolders;
    }

    public BaseDataHolder getDataHolder(String tag) {
        for (BaseDataHolder dataHolder : dataHolders) {
            if (dataHolder.getTag().equals(tag)) return dataHolder;
        }
        return null;
    }
}

File: app/src/main/java/com/raival/fileexplorer/activity/model/MainViewModel.kt
package com.raival.fileexplorer.activity.model

import androidx.lifecycle.ViewModel
import com.raival.fileexplorer.tab.BaseDataHolder
import com.raival.fileexplorer.tab.file.model.Task

class MainViewModel : ViewModel() {
    @JvmField
    val tasks = arrayListOf<Task>()
    private val dataHolders: MutableList<BaseDataHolder> = arrayListOf()

    fun addDataHolder(dataHolder: BaseDataHolder) {
        dataHolders.add(dataHolder)
    }

    fun getDataHolders(): ArrayList<BaseDataHolder> {
        return dataHolders as ArrayList<BaseDataHolder>
    }

    fun getDataHolder(tag: String): BaseDataHolder? {
        for (dataHolder in dataHolders) {
            if (dataHolder.tag == tag) return dataHolder
        }
        return null
    }
}